src/ui/public/pattern_checker/pattern_checker.js

```diff
@@ -0,0 +1,56 @@
+import uiModules from 'ui/modules';
+import template from './pattern_checker.html';
+import './pattern_checker.less';
+
+const module = uiModules.get('kibana');
+
+module.directive('patternChecker', function () {
+  return {
+    restrict: 'E',
+    template: template,
+    controllerAs: 'checker',
+    bindToController: true,
+    scope: {
+      pattern: '=',
+      buttonLabel: '='
+    },
+    controller: function (es, Notifier, $scope, $timeout) {
+      let validationTimeout;
+      this.isValidating = false;
+
+      var notify = new Notifier({
+        location: 'Add Data'
+      });
+
+      this.toggleValidation = () => {
+        if (!this.isValidating) {
+          this.isValidating = true;
+          this.validateInstall();
+        }
+        else {
+          $timeout.cancel(validationTimeout);
+          this.isValidating = false;
+          this.validationResults = '';
+        }
+      };
+
+      this.validateInstall = () => {
+        es.count({
```

Comment:
We want to eventually get the es client out of the frontend.
Can we make this a REST endpoint?